bench/rttanalysis: TestBenchmarkExpectation failed

bench/rttanalysis.TestBenchmarkExpectation [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20653154?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20653154?buildTab=artifacts#/) on master @ [c4dbc61b5588a3c607a5c7ee9dcf38be7750787f](https://github.com/cockroachdb/cockroach/commits/c4dbc61b5588a3c607a5c7ee9dcf38be7750787f):


```
    validate_benchmark_data.go:120: success: expected SystemDatabaseQueries/select_system.users_without_schema_Name to perform KV lookups in [1, 1], got 1
    validate_benchmark_data.go:120: success: expected GrantRole/grant_1_role to perform KV lookups in [12, 12], got 12
    validate_benchmark_data.go:120: success: expected Truncate/truncate_2_column_1_rows to perform KV lookups in [17, 17], got 17
    validate_benchmark_data.go:120: success: expected Truncate/truncate_1_column_0_rows to perform KV lookups in [18, 18], got 18
    validate_benchmark_data.go:120: success: expected DropView/drop_1_view to perform KV lookups in [14, 14], got 14
    validate_benchmark_data.go:120: success: expected DropView/drop_2_views to perform KV lookups in [15, 15], got 15
    validate_benchmark_data.go:120: success: expected RevokeRole/revoke_1_role to perform KV lookups in [10, 10], got 10
    validate_benchmark_data.go:120: success: expected DropTable/drop_1_table to perform KV lookups in [13, 13], got 13
    validate_benchmark_data.go:120: success: expected Jobs/non_admin_crdb_internal.system_jobs to perform KV lookups in [2, 2], got 2
    validate_benchmark_data.go:120: success: expected Grant/grant_all_on_1_table to perform KV lookups in [8, 8], got 8
    validate_benchmark_data.go:120: success: expected VirtualTableQueries/select_crdb_internal.invalid_objects_with_1_fk to perform KV lookups in [2, 2], got 2
    validate_benchmark_data.go:120: success: expected AlterTableDropColumn/alter_table_drop_1_column to perform KV lookups in [22, 22], got 22
    validate_benchmark_data.go:120: success: expected Jobs/resume_job to perform KV lookups in [9, 12], got 9
    validate_benchmark_data.go:120: success: expected VirtualTableQueries/virtual_table_cache_with_point_lookups to perform KV lookups in [5, 5], got 5
    validate_benchmark_data.go:120: success: expected Jobs/crdb_internal.system_jobs to perform KV lookups in [3, 3], got 3
    validate_benchmark_data.go:120: success: expected VirtualTableQueries/virtual_table_cache_with_schema_change to perform KV lookups in [10, 10], got 10
    validate_benchmark_data.go:120: success: expected Discard/DISCARD_ALL,_no_tables to perform KV lookups in [0, 0], got 0
    validate_benchmark_data.go:120: success: expected DropTable/drop_2_tables to perform KV lookups in [15, 15], got 15
    validate_benchmark_data.go:120: success: expected AlterTableConfigureZone/alter_table_configure_zone_5_replicas to perform KV lookups in [14, 14], got 14
    validate_benchmark_data.go:120: success: expected Jobs/jobs_page_default to perform KV lookups in [3, 5], got 3
    validate_benchmark_data.go:120: success: expected Jobs/show_job to perform KV lookups in [3, 3], got 3
    validate_benchmark_data.go:120: success: expected Jobs/cancel_job to perform KV lookups in [9, 12], got 9
    validate_benchmark_data.go:120: success: expected Jobs/pause_job to perform KV lookups in [9, 12], got 11
    validate_benchmark_data.go:120: success: expected Grant/grant_all_on_2_tables to perform KV lookups in [8, 8], got 8
    validate_benchmark_data.go:120: success: expected AlterTableDropColumn/alter_table_drop_2_columns to perform KV lookups in [22, 22], got 22
    validate_benchmark_data.go:120: success: expected Jobs/non_admin_show_jobs to perform KV lookups in [2, 2], got 2
    validate_benchmark_data.go:120: success: expected Truncate/truncate_1_column_1_row to perform KV lookups in [17, 17], got 17
    validate_benchmark_data.go:120: success: expected ORMQueries/prisma_types_16 to perform KV lookups in [5, 5], got 5
    validate_benchmark_data.go:120: success: expected Jobs/jobs_page_latest_50 to perform KV lookups in [3, 3], got 3
    validate_benchmark_data.go:120: success: expected Truncate/truncate_1_column_2_rows to perform KV lookups in [17, 17], got 17
    validate_benchmark_data.go:120: success: expected Jobs/jobs_page_type_filtered to perform KV lookups in [3, 3], got 3
    validate_benchmark_data.go:120: success: expected Truncate/truncate_2_column_2_rows to perform KV lookups in [17, 17], got 17
    validate_benchmark_data.go:120: success: expected Truncate/truncate_2_column_0_rows to perform KV lookups in [18, 18], got 18
    validate_benchmark_data.go:120: success: expected UseManyRoles/use_50_roles to perform KV lookups in [2, 2], got 2
    validate_benchmark_data.go:120: success: expected Jobs/jobs_page_type_filtered_no_matches to perform KV lookups in [1, 3], got 1
    validate_benchmark_data.go:120: success: expected Jobs/show_jobs to perform KV lookups in [3, 5], got 3
    validate_benchmark_data.go:120: success: expected VirtualTableQueries/show_create_all_routines to perform KV lookups in [0, 0], got 0
    validate_benchmark_data.go:120: success: expected VirtualTableQueries/show_create_all_triggers to perform KV lookups in [0, 0], got 0
    validate_benchmark_data.go:73: see the -rewrite flag to re-run the benchmarks and adjust the expectations
    validate_benchmark_data.go:74: usage: ./dev test --rewrite pkg/bench/rttanalysis -f=TestBenchmarkExpectation/ --test-args '-rewrite-iterations=N'
    validate_benchmark_data.go:107: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/3e7abc59ef4f9b4e3bad6013583e698d/logTestBenchmarkExpectation468270393
--- FAIL: TestBenchmarkExpectation (838.18s)
--- FAIL: TestBenchmarkExpectation (839.72s)
=== RUN   TestBenchmarkExpectation/ORMQueries/liquibase_migrations_on_multiple_dbs
    rtt_analysis_bench.go:210: error executing query="DROP DATABASE d0;\nDROP DATABASE d1;\nDROP DATABASE d2;\nDROP DATABASE d3;\nDROP DATABASE d4;\nDROP DATABASE d5;\nDROP DATABASE d6;\nDROP DATABASE d7;\nDROP DATABASE d8;\nDROP DATABASE d9;\nDROP DATABASE d10;\nDROP DATABASE d11;\nDROP DATABASE d12;\nDROP DATABASE d13;\nDROP DATABASE d14;\n" args=[]: pq: restart transaction: error executing PreCommitPhase stage 2 of 2 with 1657 MutationType ops: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_CLIENT_REJECT): "sql txn" meta={id=9606c6dd key=/Table/3/1/317/2/1 iso=Serializable pri=0.00044121 epo=0 ts=1761509284.673257400,2 min=1761509267.006555309,0 seq=84} lock=true stat=PENDING rts=1761509267.006555309,0 gul=1761509267.506555309,0 obs={n1@1761509267.006555309,0}
        HINT: See: https://www.cockroachlabs.com/docs/v25.4/transaction-retry-error-reference.html#abort_reason_client_reject
--- FAIL: TestBenchmarkExpectation/ORMQueries/liquibase_migrations_on_multiple_dbs (358.16s)
=== RUN   TestBenchmarkExpectation/ORMQueries
--- FAIL: TestBenchmarkExpectation/ORMQueries (613.09s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #155606 bench/rttanalysis: TestBenchmarkExpectation failed [Jobs/jobs_page_latest_50: got 5.5, expected 8] [A-jobs C-test-failure O-robot T-jobs branch-release-24.1]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBenchmarkExpectation.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-55871

--- a/pkg/bench/rttanalysis/orm_queries_bench_test.go
+++ b/pkg/bench/rttanalysis/orm_queries_bench_test.go
@@ -946,8 +946,8 @@ ORDER BY
 		{
 			Name: `liquibase migrations on multiple dbs`,
 			// 15 databases, each with 40 tables.
-			Setup: liquibaseSetup,
-			Reset: liquibaseReset,
+			SetupEx: liquibaseSetup,
+			ResetEx: liquibaseReset,
 			Stmt: `SELECT
   NULL AS table_cat,
   n.nspname AS table_schem,
@@ -1067,16 +1067,18 @@ func buildNTypes(n int) string {
 	return b.String()
 }
 
-func buildNDatabasesWithMTables(amtDbs int, amtTbls int) (string, string) {
-	b := strings.Builder{}
-	reset := strings.Builder{}
+func buildNDatabasesWithMTables(amtDbs int, amtTbls int) ([]string, []string) {
+	setupEx := make([]string, amtDbs)
+	resetEx := make([]string, amtDbs)
 	tbls := buildNTables(amtTbls)
 	for i := 0; i < amtDbs; i++ {
 		db := fmt.Sprintf("d%d", i)
+		b := strings.Builder{}
 		b.WriteString(fmt.Sprintf("CREATE DATABASE IF NOT EXISTS %s;\n", db))
-		reset.WriteString(fmt.Sprintf("DROP DATABASE %s;\n", db))
 		b.WriteString(fmt.Sprintf("USE %s;\n", db))
 		b.WriteString(tbls)
+		setupEx[i] = b.String()
+		resetEx[i] = fmt.Sprintf("DROP DATABASE %s", db)
 	}
-	return b.String(), reset.String()
+	return setupEx, resetEx
 }

